### PR TITLE
Nest Middlewares - Passport

### PR DESCRIPTION
**Pull request by revant**
instead of passport initialize(), session() in main.ts

these middlewares can be used on any controller, e.g. AppController.

### CODE REVIEW

File: packages/passport/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+```ts
+import { Module, NestModule, MiddlewareConsumer } from '@nestjs/common';
+import { PassportInitializeMiddleware, PassportSessionMiddleware } from '@nest-middlewares/passport';
+import { AppController } from './app.controller';
+
+@Module(...)
+export class AppModule implements NestModule {
+    configure(consumer: MiddlewaresConsumer) {
+        consumer
+        	.apply(PassportInitializeMiddleware)
```
Comment (wbhob) [inline packages/passport/README.md:26]: I don't know if this config will work (I think you need to reference consumer twice, as forRoutes doesn't apply anything. Also, forRoutes takes an object as far as I know).

**Review by wbhob — CHANGES_REQUESTED** [review]
Overall, looks really good. Thanks for your contribution! If you could also add some tests to make sure everything is initializing correctly, that would be a big help.

Also, you'll need to update the lockfile for the CI to turn green (rm -r node_modules; rm package-lock.json; npm install).

If you don't know how to do this, I can give you a hand with it. :) cheers!

File: packages/passport/README.md (hunk 1 above)
Comment (revant) [reply packages/passport/README.md:26]: I'm using it here [[https://gitlab.com/revant.one/identity-provider](https://gitlab.com/revant.one/identity-provider)] and it is working, i.e Session getting created and stored into Typeorm session store, Browser shows logged in user, `req.user` persists until logout.

I faced another problem with this setup, I can't run `test:e2e`, it fails with

```
$ yarn test:e2e
yarn run v1.7.0
$ jest --config ./test/jest-e2e.json --forceExit
 FAIL  test/app.e2e-spec.ts
  AppController (e2e)
    ✕ /GET / (25ms)

  ● AppController (e2e) › /GET /

    TypeError: Cannot read property 'initialize' of undefined
      
      at Object.<anonymous>.Authenticator.initialize (../node_modules/passport/lib/authenticator.js:133:26)
      at PassportInitializeMiddleware.resolve (../src/middlewares/passport.middlewares.ts:6:29)
      at MiddlewareModule.bindHandler (../node_modules/@nestjs/core/middleware/middleware-module.js:80:34)
      at Promise.all.middlewareCollection.map (../node_modules/@nestjs/core/middleware/middleware-module.js:68:24)
          at Array.map (<anonymous>)
      at MiddlewareModule.registerRouteMiddleware (../node_modules/@nestjs/core/middleware/middleware-module.js:61:48)
      at Promise.all.forRoutes.map (../node_modules/@nestjs/core/middleware/middleware-module.js:56:24)
          at Array.map (<anonymous>)
      at MiddlewareModule.registerMiddlewareConfig (../node_modules/@nestjs/core/middleware/middleware-module.js:55:37)
      at middlewareConfig.map (../node_modules/@nestjs/core/middleware/middleware-module.js:47:24)
          at Array.map (<anonymous>)
      at registerAllConfigs (../node_modules/@nestjs/core/middleware/middleware-module.js:46:83)
      at Promise.all.map (../node_modules/@nestjs/core/middleware/middleware-module.js:50:31)
          at Array.map (<anonymous>)
      at MiddlewareModule.registerMiddleware (../node_modules/@nestjs/core/middleware/middleware-module.js:49:50)
      at NestApplication.registerMiddleware (../node_modules/@nestjs/core/nest-application.js:251:37)
      at NestApplication.registerRouter (../node_modules/@nestjs/core/nest-application.js:117:20)
      at NestApplication.init (../node_modules/@nestjs/core/nest-application.js:88:20)
          at <anonymous>
      at process._tickCallback (../internal/process/next_tick.js:188:7)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        3.563s
Ran all test suites.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
Anything I missed?

Comment (wbhob) [reply packages/passport/README.md:26]: seems like you haven't installed passport to your local.

also, i was talking about copying the simple test file to make sure it exists. I haven't been able to do e2e tests on this project yet

Comment (revant) [reply packages/passport/README.md:26]: solved it,

`process.env.NODE_ENV==='test'` behaves differently.

https://gitlab.com/revant.one/identity-provider/blob/fe83b64aa56ae08e59cbda7cd55adfc749b14a50/src/auth/auth.module.ts#L25

Comment (wbhob) [reply packages/passport/README.md:26]: did you make the changes I requested above?

Comment (revant) [reply packages/passport/README.md:26]: > did you make the changes I requested above?

No, I'll get back.
Trying to understand things.

> Also, you'll need to update the lockfile for the CI to turn green (rm -r node_modules; rm package-lock.json; npm install).

package-lock.json I understood. 

> if you could also add some tests to make sure everything is initializing correctly, that would be a big help.

I'll try this. I can't promise anything.

> If you don't know how to do this, I can give you a hand with it. :) cheers!

That will be great!

**Comment by revant** [comment]
package json didn't help green tick :(

locally tests ran successfully
<details>

```
 17:02:37  revant@revant-laptop  ~/tmp/nest-middlewares  ⬡ v10.5.0   master ✔  38s 
$ npm run test     

> nest-middlewares@5.0.1 test /home/revant/tmp/nest-middlewares
> nyc --require ts-node/register mocha packages/**/*.spec.ts --reporter spec



  CompressionMiddleware
    middleware configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw an error for not being configured

  ConnectRidMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should return a middleware from calling resolve

  ConnectTimeoutMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw if improperly configured

  CookieParserMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    properly configured with only secret
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should return a middleware from calling resolve

  CookieSessionMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw error when not given keys

  CorsMiddleware
    middleware configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw an error for not being configured

  CsurfMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should return a middleware from calling resolve

  ErrorHandlerMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should return a middleware from calling resolve

  ExpressBearerTokenMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should return a middleware from calling resolve

  ExpressSessionMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
express-session deprecated undefined resave option; provide resave option packages/express-session/index.ts:19:20
express-session deprecated undefined saveUninitialized option; provide saveUninitialized option packages/express-session/index.ts:19:20
      ✓ should return a middleware from calling resolve
    not configured
express-session deprecated undefined resave option; provide resave option packages/express-session/index.ts:22:20
express-session deprecated undefined saveUninitialized option; provide saveUninitialized option packages/express-session/index.ts:22:20
express-session deprecated req.secret; provide secret option packages/express-session/index.ts:22:20
      ✓ should return a middleware from calling resolve

  HelmetContentSecurityPolicyMiddleware
    middleware properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw an error for not being configured

  HelmetDnsPrefetchControlMiddleware
    middleware configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw an error for not being configured

  HelmetExpectCtMiddleware
    middleware configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw an error for not being configured

  HelmetFrameguardMiddleware
    middleware configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw an error for not being configured

  HelmetHidePoweredByMiddleware
    middleware configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw an error for not being configured

  HelmetHpkpMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    improperly configured
      ✓ should throw if improperly configured

  HelmetHstsMiddleware
    middleware configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw an error for not being configured

  HelmetIeNoOpenMiddleware
    ✓ should be defined
    ✓ should have a function called resolve
    ✓ should return a middleware from calling resolve

  HelmetMiddleware
    middleware configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw an error for not being configured

  HelmetNoCacheMiddleware
    ✓ should be defined
    ✓ should have a function called resolve
    ✓ should return a middleware from calling resolve

  HelmetNoSniffMiddleware
    ✓ should be defined
    ✓ should have a function called resolve
    ✓ should return a middleware from calling resolve

  HelmetReferrerPolicyMiddleware
    middleware configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw an error for not being configured

  HelmetXssFilterMiddleware
    middleware configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw an error for not being configured

  MethodOverrideMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should return a middleware from calling resolve

  MorganMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw when not passed a format
    token generation
      ✓ should create a token when calling token()

  PassportAuthenticateMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should return a middleware from calling resolve

  PassportInitializeMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should return a middleware from calling resolve

  PassportSessionMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should return a middleware from calling resolve

  ResponseTimeMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should return a middleware from calling resolve

  ServeFaviconMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw for being improperly configured

  ServeIndexMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw for being improperly configured

  ServeStaticMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw if a path is not specified

  VhostMiddleware
    properly configured
      ✓ should be defined
      ✓ should have a function called resolve
      ✓ should return a middleware from calling resolve
    not configured
      ✓ should throw if a path and handler are not specified


  131 passing (78ms)


=============================== Coverage summary ===============================
Statements   : Unknown% ( 0/0 )
Branches     : Unknown% ( 0/0 )
Functions    : Unknown% ( 0/0 )
Lines        : Unknown% ( 0/0 )
================================================================================
```

</details>

**Review by wbhob — APPROVED** [review]

**Comment by wbhob** [comment]
The check is for Node.JS 6. I'll drop support and add a note in the README. Thanks for the PR and your patience!

**Pull request merged**